### ISSUE TITLE
Exception being thrown due to com.intellij.openapi.components.BaseState.map$default being removed in 2019.2

### ISSUE THREAD

**Issue by G-Rath** [aws/aws-toolkit-jetbrains#1112]
[The `com.intellij.openapi.components.BaseState.map$default(BaseState, Map, int, Object)` method has been removed in 2019.2, in favor of `com.intellij.openapi.components.BaseState.map()`](http://www.jetbrains.org/intellij/sdk/docs/reference_guide/api_changes/api_changes_list_2019.html#changes-in-intellij-platform-20192).

As such, it's causing an exception to be thrown whenever I right-click on some things:

```
update failed for AnAction(com.intellij.execution.actions.RunContextAction) with ID=RunClass

java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:116)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:469)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:402)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:376)
	at com.intellij.execution.configurations.RunConfigurationBase.createOptions(RunConfigurationBase.java:57)
	at com.intellij.execution.configurations.RunConfigurationBase.<init>(RunConfigurationBase.java:52)
	at com.intellij.execution.configurations.LocatableConfigurationBase.<init>(LocatableConfigurationBase.java:19)
	at software.aws.toolkits.jetbrains.services.lambda.execution.LambdaRunConfigurationBase.<init>(LambdaRunConfigurationBase.kt:27)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfiguration.<init>(LocalLamdaRunConfiguration.kt:50)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfigurationFactory.createTemplateConfiguration(LocalLamdaRunConfiguration.kt:42)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaRunConfigurationFactory.createTemplateConfiguration(LocalLamdaRunConfiguration.kt:41)
	at com.intellij.execution.configurations.ConfigurationFactory.createTemplateConfiguration(ConfigurationFactory.java:65)
	at com.intellij.execution.impl.RunManagerImpl.createTemplateSettings$intellij_platform_lang_impl(RunManagerImpl.kt:301)
	at com.intellij.execution.impl.RunManagerImpl.getConfigurationTemplate(RunManagerImpl.kt:293)
	at com.intellij.execution.impl.RunManagerImpl.createConfiguration(RunManagerImpl.kt:215)
	at com.intellij.execution.actions.RunConfigurationProducer.cloneTemplateConfigurationStatic(RunConfigurationProducer.java:257)
	at com.intellij.execution.actions.RunConfigurationProducer.cloneTemplateConfiguration(RunConfigurationProducer.java:247)
	at com.intellij.execution.actions.RunConfigurationProducer.createConfigurationFromContext(RunConfigurationProducer.java:96)
	at com.intellij.execution.actions.RunConfigurationProducer.findOrCreateConfigurationFromContext(RunConfigurationProducer.java:197)
	at com.intellij.execution.actions.PreferredProducerFind.doGetConfigurationsFromContext(PreferredProducerFind.java:110)
	at com.intellij.execution.actions.PreferredProducerFind.getConfigurationsFromContext(PreferredProducerFind.java:93)
	at com.intellij.execution.actions.PreferredProducerFind.findConfigurationFromContext(PreferredProducerFind.java:135)
	at com.intellij.execution.actions.PreferredProducerFind.createConfiguration(PreferredProducerFind.java:32)
	at com.intellij.execution.actions.ConfigurationContext.createConfiguration(ConfigurationContext.java:147)
	at com.intellij.execution.actions.ConfigurationContext.getConfiguration(ConfigurationContext.java:138)
	at com.intellij.execution.actions.BaseRunConfigurationAction.update(BaseRunConfigurationAction.java:179)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:176)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:378)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$null$1(ActionUpdater.java:78)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.callAction(ActionUpdater.java:106)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$new$2(ActionUpdater.java:79)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.update(ActionUpdater.java:364)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:231)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:215)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1609)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:258)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:215)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1609)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:258)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:215)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1609)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:258)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:215)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1609)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:258)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:215)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1609)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:215)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:139)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:131)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$expandActionGroupWithTimeout$11(ActionUpdater.java:148)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$3(CoreProgressManager.java:185)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:185)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.withTimeout(ProgressIndicatorUtils.java:260)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroupWithTimeout(ActionUpdater.java:147)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:66)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:114)
	at com.intellij.ui.PopupHandler$2.invokePopup(PopupHandler.java:88)
	at com.intellij.ui.PopupHandler.mouseReleased(PopupHandler.java:48)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(Unknown Source)
	at java.desktop/java.awt.Component.processMouseEvent(Unknown Source)
	at java.desktop/javax.swing.JComponent.processMouseEvent(Unknown Source)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:395)
	at com.intellij.ide.dnd.aware.DnDAwareTree.processMouseEvent(DnDAwareTree.java:44)
	at java.desktop/java.awt.Component.processEvent(Unknown Source)
	at java.desktop/java.awt.Container.processEvent(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.Component.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$5.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:873)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:818)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:461)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:704)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:460)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedConstructorAccessor1145.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at com.intellij.util.ReflectionUtil.createAsDataClass(ReflectionUtil.java:445)
	... 105 more
Caused by: java.lang.NoSuchMethodError: com.intellij.openapi.components.BaseState.map$default(Lcom/intellij/openapi/components/BaseState;Ljava/util/Map;ILjava/lang/Object;)Lcom/intellij/openapi/components/StoredPropertyBase;
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.FunctionOptions.<init>(LocalLambdaOptions.kt:26)
	at software.aws.toolkits.jetbrains.services.lambda.execution.local.LocalLambdaOptions.<init>(LocalLambdaOptions.kt:14)
	... 109 more
```

**Your Environment**
IntelliJ IDEA 2019.2 (Ultimate Edition)
Build #IU-192.5728.98, built on July 23, 2019
Licensed to Gareth Jones
Subscription is active until June 22, 2020
Runtime version: 11.0.2+9-b226.7 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1981M
Cores: 8

**Comment by G-Rath** [comment]
Looking around, I think this was fixed a month ago as part of this [commit](https://github.com/aws/aws-toolkit-jetbrains/commit/32656797418df3fb66271883117767534a694060)

**Comment by abrooksv** [comment]
Hey, thanks for the report. You are correct the issue is fixed and checked into master. Our integration and ui tests revealed a few more issues with 2019.2 and I have checked in the fixes for those issues. We are going as quick as possible to get 1.5 out the door to get these fixes out.

**Comment by G-Rath** [comment]
No problem - I've switched to the EAP version for now. When I get around to restarting my IDE, I'll let you know if there's any other errors.

It seems that this particular removal has hit a number of plugins hard 😬

**Comment by abrooksv** [comment]
We have released 1.5, and is pending JetBrains' approval on the market place
